src/GameSimulator/index.js

```diff
@@ -1,74 +1,79 @@
 import Car from '../Car/index.js';
 import RacingGame from '../RacingGame/index.js';
-import { getUserInputByQuestion } from '../utils/getUserInputByQuestion.js';
-import { MAX_ROUNDS } from './constants.js';
 
 class GameSimulator {
   #racingGame;
-  #messageViewer;
+  #gameViewer;
+  #gameInput;
+  #carNames;
   #maxRounds;
+  checkCanMoveForward;
 
-  constructor(messageViewer, maxRounds = MAX_ROUNDS) {
-    this.#messageViewer = messageViewer;
-    this.#maxRounds = maxRounds;
+  constructor(gameViewer, gameInput, checkCanMoveForward) {
+    this.#gameViewer = gameViewer;
+    this.#gameInput = gameInput;
+    this.checkCanMoveForward = checkCanMoveForward;
   }
 
   async setRacingGame() {
-    const inputString = await getUserInputByQuestion(
-      '경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).'
-    );
-    const carNames = inputString.split(',');
+    if (!this.#carNames) {
+      await this.setCarNames();
+    }
 
-    this.#racingGame = new RacingGame(carNames.map(Car.of));
-  }
+    if (!this.#maxRounds) {
+      await this.setMaxRounds();
+    }
 
-  runRound() {
-    this.#racingGame.runRound();
-    this.#racingGame.getCars().forEach((car) => {
-      this.printCarStatus(car.getName(), car.getDistanceDriven());
-    });
+    this.#racingGame = new RacingGame(
+      this.#carNames.map(Car.of),
+      this.#maxRounds
+    );
```

Comment:
```suggestion
    this.#racingGame = new RacingGame(
      this.#carNames,
      this.#maxRounds
    );
```
Car 생성은 game에게 위임해도 무방해보여요!